[#37018229] Modified spec and resource last page handling

ke4 opened the pull request.

Commit:
[#37018229] Modified spec and resource last page handling

maxigit commented on spec/lims-api/core_resource_page_spec.rb:
```diff
@@ -175,7 +175,7 @@ def content_to_stream(s)
           }  
         end
         context "last page" do
-          include_context "page", 5
```
This should still be there, as page=-1 is only an alias to the last page.
(see line 181 :read => /models/page=5"

page 5 and -1 needs to be tested

maxigit commented on spec/lims-api/core_resource_page_spec.rb:
```diff
@@ -175,7 +175,7 @@ def content_to_stream(s)
+          include_context "page", -1
           it_behaves_like "page resource", {
             "actions" => {
             "read" => "/models/page=5",
```
This is correct and should not be changed to -1.

maxigit commented:
Please readd test page=5

Commit:
[#37018229] Reinstated direct test of last page

We left the test of the alias in place, as this had been failing
prior to our changes.
